sdk/python/feast/entity.py

```diff
@@ -78,7 +74,8 @@ def __init__(
         self._created_timestamp: Optional[datetime] = None
         self._last_updated_timestamp: Optional[datetime] = None
 
-        self.defined_in = get_calling_file_name(inspect.stack())
+    def __hash__(self) -> int:
+        return hash(self.name)
```

Comment:
How would we detect and raise an exception when users provide two objects with the same name that are different?

Reply:
Having two entities with the same name is a bug though, isn't it?

Reply:
I don't think we support any FV/FT/entity/FS having the same name right now anyway

Reply:
Okay, I checked the code and we verify unique names for FVs but not other objects. I don't know what the behaviour would be for `FeatureStore.get_feature_view` if names can be duplicated. Plus, feature_refs would be busted (in the case of feature views and feature tables). wdyt?

Reply:
My earlier comments completely misunderstood your point -fixing using `id()`.